engine/graphics/src/null/graphics_null.cpp

```diff
@@ -61,6 +61,26 @@ namespace dmGraphics
 
     DM_REGISTER_GRAPHICS_ADAPTER(GraphicsAdapterNull, &g_null_adapter, NullIsSupported, NullRegisterFunctionTable, g_null_adapter_priority);
 
+    struct AssetContainerLock
+    {
+        AssetContainerLock(dmMutex::HMutex mutex)
+        : m_Mutex(mutex)
+        {
+            if (m_Mutex)
+            {
+                dmMutex::Lock(m_Mutex);
+            }
+        }
+        ~AssetContainerLock()
+        {
+            if (m_Mutex)
+            {
+                dmMutex::Unlock(m_Mutex);
+            }
+        }
+        dmMutex::HMutex m_Mutex;
+    };
```

Comment:
To remove any ambiguity about the scopes and potential mismatches (while I was debugging), I created this scoped lock.